core/rint/src/TTabCom.cxx

```diff
@@ -868,10 +869,17 @@ TString TTabCom::DetermineClass(const char varName[])
    type.ReadToDelim(file1, ')');
    IfDebug(std::cerr << type << std::endl);
 
-   // new version of CINT returns: "class TClassName*const)0x12345"
-   // so we have to strip off "const"
-   if (type.EndsWith("const"))
+   // new version of Cling returns: "qualifiers TClassName *const) 0x12345"
+   // we start stripping off right "const"
+   if (type.EndsWith(" *const"))
       type.Remove(type.Length() - 5);
```

Comment:
Note that this is still only stripping off the last five characters, ie `const`. Why do we need to check ` *` in the beginning?

Reply:
To avoid problems if one wants to define a class name called e.g. "Reconst", so looking for the space ensures that it's a qualifier rather than the end of the classname.
Then, we do not want to remove the * from the type, since we want to know if it's a pointer or not, to verify if it should go with -> or .

Reply:
Probably one could also just check for `*const` but I thought it was more clean / understandable for the developers reading this code to also check for the space.